source/simulator/core.cc

```diff
@@ -1678,19 +1678,38 @@ namespace aspect
             free_surface->execute ();
 
           assemble_advection_system (AdvectionField::temperature());
-          build_advection_preconditioner(AdvectionField::temperature(),
-                                         T_preconditioner);
-          solve_advection(AdvectionField::temperature());
+          if (system_rhs.block(AdvectionField::temperature().block_index(introspection)).l2_norm() <= std::numeric_limits<double>::min())
```

Comment:
This should just compare to numeric zero, for two reasons. 

First, `x <= numeric_limits::min()` is true if either `x==numeric_limits::min()` (which is a floating point comparison for equality that is unlikely to ever be true), or if `x==0`. The definition of `numeric_limit::min()` is so that the condition `x < numeric_limit::min()` is never true.

Second, `rhs.l2_norm()` has physical units, whereas `numeric_limits::min()` does not, and so a comparison cannot be correct without a scaling factor.